lib/suggest/suggest.ts

```diff
@@ -0,0 +1,193 @@
+// Credits go to Liam's Periodic Notes Plugin: https://github.com/liamcain/obsidian-periodic-notes
+
+import { ISuggestOwner, Scope } from 'obsidian';
+import { createPopper, Instance as PopperInstance } from '@popperjs/core';
+
+const wrapAround = (value: number, size: number): number => {
+  return ((value % size) + size) % size;
+};
+
+class Suggest<T> {
+  private owner: ISuggestOwner<T>;
+  private values: T[];
+  private suggestions: HTMLDivElement[];
+  private selectedItem: number;
+  private containerEl: HTMLElement;
+
+  constructor(owner: ISuggestOwner<T>, containerEl: HTMLElement, scope: Scope) {
+    this.owner = owner;
+    this.containerEl = containerEl;
+
+    containerEl.on(
+      'click',
+      '.suggestion-item',
+      this.onSuggestionClick.bind(this)
+    );
+    containerEl.on(
+      'mousemove',
+      '.suggestion-item',
+      this.onSuggestionMouseover.bind(this)
+    );
+
+    scope.register([], 'ArrowUp', (event) => {
+      if (!event.isComposing) {
+        this.setSelectedItem(this.selectedItem - 1, true);
+        return false;
+      }
+    });
+
+    scope.register([], 'ArrowDown', (event) => {
+      if (!event.isComposing) {
+        this.setSelectedItem(this.selectedItem + 1, true);
+        return false;
+      }
+    });
+
+    scope.register([], 'Enter', (event) => {
+      if (!event.isComposing) {
+        this.useSelectedItem(event);
+        return false;
+      }
+    });
+  }
+
+  onSuggestionClick(event: MouseEvent, el: HTMLDivElement): void {
```

Comment:
if you change this to use arrow functions you won't need to use `.bind` above. Same with the functions below